### PR TITLE
cp-270: Implement Chat Messaging Functionality

### PR DESCRIPTION
**Pull request by danilittaa**
Done: 
- A new chat window should be opened when the user clicks on the 'Add chat' icon
- User should be able to type and send a message with the help of the input field
- The sent message should be displayed on the chat page
- User should receive a reply form AI each time he sends a message
 
 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/108797614/a0537bc4-1d29-4f75-bc8c-a0e374f0aaad

### CODE REVIEW

**Review by dmytro-kucherenko — COMMENTED** [review]
Looks nice! The only you can do is adding comments for now where functionality will be soon

**Review by OlesiaTr — DISMISSED** [review]

File: mobile/src/packages/chat-messages/chat-messages-api.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  ): Promise<ChatMessageGetAllResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(ChatsApiPath.$ID_MESSAGES, { id: chatId }),
+      { method: 'GET', hasAuth: true },
+    );
+
+    return await response.json<ChatMessageGetAllResponseDto>();
+  }
+
+  public async createChatMessage(
```
Comment (v0ldemar01) [inline mobile/src/packages/chat-messages/chat-messages-api.ts]: A5 rule

File: mobile/src/screens/chat-list/chat-list.tsx [hunk 2]
```diff
@@ -43,14 +44,17 @@ const ChatList: React.FC = () => {
...
+      navigation.navigate(ChatScreenName.CHAT, { title, id: id ?? '' });
     },
     [navigation],
   );
 
   const handleRedirectToChat = useCallback(() => {
-    navigation.navigate(ChatScreenName.CHAT, { title: 'New Chat' });
+    navigation.navigate(ChatScreenName.CHAT, {
+      title: 'New Chat',
+      id: '',
```
Comment (v0ldemar01) [inline mobile/src/screens/chat-list/chat-list.tsx:56]:
```suggestion
```

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/screens/chat-list/chat-list.tsx (hunk 2 above)
Comment (danilittaa) [reply mobile/src/screens/chat-list/chat-list.tsx:56]: I need to pass id, it's mandatory; without it, I get an error

Comment (liza-veis) [reply mobile/src/screens/chat-list/chat-list.tsx:56]: it's ok in this case

**Review by liza-veis — APPROVED** [review]

**Review by v0ldemar01 — APPROVED** [review]

**Pull request merged**